feat: add messages field in the Operation Object (#870)
asyncapi/spec: spec/asyncapi.md

---|:---:|---
<a name="operationObjectAction"></a>action | `string` | **Required**. Allowed values are `send` and `receive`. Use `send` when it's expected that the application will send a message to the given [`channel`](#operationObjectChannel), and `receive` when the application should expect receiving messages from the given [`channel`](#operationObjectChannel).
<a name="operationObjectChannel"></a>channel | [Reference Object](#referenceObject) | **Required**. A `$ref` pointer to the definition of the channel in which this operation is performed. Please note the `channel` property value MUST be a [Reference Object](#referenceObject) and, therefore, MUST NOT contain a [Channel Object](#channelObject). However, it is RECOMMENDED that parsers (or other software) dereference this property for a better development experience.

> <a name="operationObjectMessages"></a>messages | [Messages Object](#messagesObject) | A map of the messages that will be processed by this operation at any time during the runtime of the application. **The map should contain a subset of messages that are defined in the corresponding channel. If the field is omitted, the operation performs an action on every message defined in the corresponding channel.**

Reviewer: I'm not sure about 
> "messages that will be processed by this operation at any time during the runtime"

But I suck at writing English so I'm struggling to come up with something much better. 

What about: 

"A map of messages that the application should send to, or be able to receive from, the channel."

or 

"A map of messages that the application must send to, or be able to receive from, the channel."

?

Reviewer 2: Another suggestion:

```suggestion
<a name="operationObjectMessages"></a>messages | [Messages Object](#messagesObject) | A map of the messages that MAY be used by this operation. **The map MUST contain a subset of the messages that are defined in the corresponding channel. If the field is omitted, the operation MAY use any of the messages defined in the corresponding channel.**
```

Author: Ok, it's ok, pushed, but we should also change the description of `channel.messages` with `MAY` word to have regularity in descriptions. Thanks!

Reviewer 2: Channel messages definition is ok as it is. In the operation, it's our application the one that's responsible so that's why we gotta be careful with using `will` instead of `MAY` or `MUST`. But in the case of channels, it may be our application or another one, that's why it's fine to use `will` there. It's not our application contract. Messages will be sent there even if our application does nothing with them.

Author: Yeah, you're right. Thanks for clarification!
